style-guide.md

```diff
@@ -0,0 +1,65 @@
+# WDL Style Guide
+
+All rules below should be followed by contributors to this repo. Pull Requests which do not conform to these specifications will be asked to change.
+
+These rules might also be enforced by a yet-to-be-written linter.
+
+## Rules
+
+- The following sections must be present and in this order: `meta`, `parameter_meta`, `input`, `command`, `output`, `runtime`
+- The `meta` section should have a `description` of the task
+  - `meta` should *not* include `author` or `email` keys
+- All inputs must have a corresponding `parameter_meta` entry
+  - These texts should be copy and pasted from other tasks with the same input when possible
+  - Inputs and parameter meta must be in the same order
+  - TODO: rework this section
+  - That order should be logical
+    - look to similar tasks/workflows for an example of proper ordering
+    - Generally speaking:
+      - Required inputs at the top
+      - "sample" files before reference files
+      - Resource configuration at the bottom
+        - memory allocations come before disk space allocations
+      - `max_retries` should be last
+      - If applicable, `detect_nproc` immediately before `max_retries`
+- Tasks with string parameters for which a limited number of choices are valid, must be documented following the template in `string_choices_task` (see `template/task-templates.wdl`)
+  - TODO: this can't be enforced by a linter. Do we keep it in the guide?
+  - they should also fail quickly with an informative error message if an invalid input is provided
+    - In most cases, just passing the parameter to the underlying tool should produce a satisfactory error, but this must be checked for each task
```

Comment:
This should probably be in our style guide, not necessarily the general guide, since it can't be enforced by a tool.